src/pages/organisationUnits/form/ImageField.tsx

```diff
@@ -0,0 +1,110 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    FileInput,
+    FileInputChangeHandler,
+    FileList,
+    FileListItem,
+    Label,
+} from '@dhis2/ui'
+import React from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+
+export function ImageField() {
+    const dataEngine = useDataEngine()
+    const fieldName = 'image'
+    const { input } = useField(fieldName)
+    const currentFile: { name?: string; id?: string } | undefined =
+        input.value || undefined
+
+    const uploadFile = async (fileToUpload: File) => {
+        const ADD_NEW_FILE_RESOURCE_MUTATION = {
+            resource: 'fileResources',
+            type: 'create',
+            data: (de: object) => de,
```

Comment:
I know this was mostly copy-pasted from another app. But we don't really need this query object, and especially no need for the callback - since it's constructed inside the handler. 

eg. 


```ts
await dataEngine.mutate({
   resource: 'fileResources',
   type: 'create',
   data: fileToUpload})
```
should suffice.

Reply:
Thank you!